examples/platforms/simulation/radio.c

```diff
@@ -827,14 +827,11 @@ void radioTransmit(struct RadioMessage *aMessage, const struct otRadioFrame *aFr
 
 void radioSendAck(void)
 {
-    if (
-#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
-        // Determine if frame pending should be set
-        (otMacFrameIsData(&sReceiveFrame) || otMacFrameIsDataRequest(&sReceiveFrame))
-#else
-        otMacFrameIsDataRequest(&sReceiveFrame)
+    if (hasFramePending(&sReceiveFrame)
+#if OPENTHREAD_CONFIG_THREAD_VERSION < OT_THREAD_VERSION_1_2
+        && otMacFrameIsDataRequest(&sReceiveFrame)
```

Comment:
In Thread spec 1.2 3.2.6.2:

> When both the polling rx-off-when-idle Thread Interface and the MAC Coordinator Thread Interface are implementing the Enhanced Frame Pending feature and the Coordinator has determined that indirect buffered data is available for  the rx-off-when-idle interface, the Coordinator SHALL set the Frame Pending subfield value to  '1' in MAC Acknowledgement frames sent in response to **MAC Data frames** from the rx-off-when-idle interface.

I think for 1.2, we still need to ensure it's a Mac data or Mac Cmd frame. We don't need to do this for Beacon right?

Reply:
Thanks for the review. Updated.

Since radio cannot do decryption without ext address to short address mapping table, I think for now, we can only make the check as close to spec as possible. I've changed the check to 2015 frame && frame is of type command.

@Irving-cl @bukepo